### ISSUE TITLE
New parser request: git ls-remote

### ISSUE THREAD

**Issue by Xaeroxe** [kellyjonbrazil/jc#315]
I'd like to be able to see the output of git ls-remote in json

**Comment by kellyjonbrazil** [comment]
Hi - thanks for the parser suggestion!

Do you think the JSON should look like key/value pairs? Seems like that would work since the first field is always unique:

```
{
  "214cd6b9e09603b3c4fa02203b24fb2bc3d4e338": "HEAD",
  "89d4df2a05afc6cc44d557d384994d7467615778": "refs/heads/dev",
  ...
}
```

**Comment by Xaeroxe** [comment]
The first output might be duplicated if the tip of a branch is also tagged. Multiple branches could also have the same tip. I like the idea, but maybe the value should be an array?

**Comment by Xaeroxe** [comment]
Actually thinking about this further, I think the key should be the value that the end user is more likely to assign semantics to, so I think the key should be the second value, and the value should be the first. This may mean occasionally having duplicate values but I think that's okay.

**Comment by kellyjonbrazil** [comment]
I could do something fancy like -r gives you an array of objects and the default view gives you the k/v pairs as you described above.

**Comment by kellyjonbrazil** [comment]
I have a working version of this parser in the `dev` branch:
https://github.com/kellyjonbrazil/jc/blob/dev/jc/parsers/git_ls_remote.py

You can test the parser by copying it to your plugin folder:

- Linux/unix: `$HOME/.local/share/jc/jcparsers`
- macOS: `$HOME/Library/Application Support/jc/jcparsers`
- Windows: `$LOCALAPPDATA\jc\jc\jcparsers`

Note that this parser requires `jc` version 1.22.1 to run. Let me know if you run into any issues!

**Comment by kellyjonbrazil** [comment]
This is what the output looks like:
```
$ git ls-remote | jc --git-ls-remote -p
{
  "HEAD": "214cd6b9e09603b3c4fa02203b24fb2bc3d4e338",
  "refs/heads/dev": "b884f6aacca39e05994596d8fdfa7e7c4f1e0389",
  "refs/heads/master": "214cd6b9e09603b3c4fa02203b24fb2bc3d4e338",
  "refs/pull/1/head": "e416c77bed1267254da972b0f95b7ff1d43fccef",
  ...
}

$ git ls-remote | jc --git-ls-remote -p -r
[
  {
    "reference": "HEAD",
    "commit": "214cd6b9e09603b3c4fa02203b24fb2bc3d4e338"
  },
  {
    "reference": "refs/heads/dev",
    "commit": "b884f6aacca39e05994596d8fdfa7e7c4f1e0389"
  },
  ...
]
```

**Comment by kellyjonbrazil** [comment]
Released in `jc` v1.22.2.
